### PR TITLE
feat: Add unique firestore database

### PR DESCRIPTION
**Pull request by arbrown**
* Changes firestore database to unique name to prevent issues on redeployment in existing project
* Adds more convenient polling for deployment rollouts in scripts
* Updates naming convention to match 'Cloud Deployment' solution name

### CODE REVIEW

File: infra/main.tf [hunk 1]
```diff
@@ -47,9 +47,10 @@ data "google_client_config" "default" {
 }
 
 locals {
-  resource_path = "resource"
+  resource_path     = "resource"
+  firestore_db_name = "large-data-sharing-${random_id.random_code.hex}"
```
Comment (davenportjw) [inline infra/main.tf:51]: udpate this to solution name for deployment

File: scripts/deploy-v1.sh [hunk 2]
```diff
@@ -70,6 +77,6 @@ echo -e "\n--------------------------------------------------------- "
 # Get Cloud Load Balancer configs
 FORWARDING_RULE_NAME="cloud-deployment-gke-golang"
 FORWARDING_RULE_IP="$(gcloud compute forwarding-rules list --filter="${FORWARDING_RULE_NAME}" --format="value(IP_ADDRESS)")"
-echo "V1 version was deployed successfully!
-access the web UI home page through the external load balancer IP: http://${FORWARDING_RULE_IP}/"
+kubectl rollout status deployment "${PROJECT_ID}"-lds-deployment-"${REGION}" -n "${NAMESPACE}"
```
Comment (davenportjw) [inline scripts/deploy-v1.sh:80]: lds reference here might want to change

**Review by davenportjw — APPROVED** [review]
code looks good, there are some env variables and references to the go solution in here, if that's intentional great otherwise may want ot replace with something related to this solution.

File: scripts/deploy-v1.sh (hunk 2 above)
Comment (arbrown) [reply scripts/deploy-v1.sh:80]: This is intentional.  It is running the lds backend just on GKE.  We can tidy up names later, but for now, this is how the deployment is actually named.

File: infra/main.tf (hunk 1 above)
Comment (arbrown) [reply infra/main.tf]: Ok, I updated all the lds references aside from the ones in the go api.

**Pull request merged**